test/IntegrationTests/AspNetTests.cs

```diff
@@ -65,5 +75,51 @@ public async Task SubmitsTraces()
 
         Assert.True(spans.Count >= 1, $"Expecting at least 1 span, only received {spans.Count}");
     }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    [Trait("Containers", "Windows")]
+    public async Task SubmitMetrics()
+    {
+        var collectorPort = TcpPortProvider.GetOpenPort();
+        var webPort = TcpPortProvider.GetOpenPort();
+        const int expectedMetricRequests = 1;
+
+        var testSettings = new TestSettings
+        {
+            MetricsSettings = new MetricsSettings { Port = collectorPort },
+        };
+
+        // Using "*" as host requires Administrator. This is needed to make the mock collector endpoint
+        // accessible to the Windows docker container where the test application is executed by binding
+        // the endpoint to all network interfaces. In order to do that it is necessary to open the port
+        // on the firewall.
+        using var fwPort = FirewallHelper.OpenWinPort(collectorPort, Output);
+        using var collector = new MockCollector(Output, collectorPort, host: "*");
+        using var container = await StartContainerAsync(testSettings, webPort);
+
+        var client = new HttpClient();
+
+        var response = await client.GetAsync($"http://localhost:{webPort}");
+        var content = await response.Content.ReadAsStringAsync();
+
+        Output.WriteLine("Sample response:");
+        Output.WriteLine(content);
+
+        var metricRequests = collector.WaitForMetrics(expectedMetricRequests, TimeSpan.FromSeconds(5));
+
+        using (new AssertionScope())
+        {
+            metricRequests.Count.Should().BeGreaterThanOrEqualTo(expectedMetricRequests);
+            var resourceMetrics = metricRequests.LastOrDefault().ResourceMetrics.Single();
+
+            var expectedServiceNameAttribute = new KeyValue { Key = "service.name", Value = new AnyValue { StringValue = "TestApplication.AspNet" } };
+            resourceMetrics.Resource.Attributes.Should().ContainEquivalentOf(expectedServiceNameAttribute);
+            var aspnetScope = resourceMetrics.ScopeMetrics.Single(rm => rm.Scope.Name.Equals("OpenTelemetry.Instrumentation.AspNet", StringComparison.OrdinalIgnoreCase));
+
+            var httpServerDurationMetric = aspnetScope.Metrics.FirstOrDefault(m => m.Name.Equals("http.server.duration", StringComparison.OrdinalIgnoreCase));
+            httpServerDurationMetric.Should().NotBeNull();
```

Comment:
Try something like:

```csharp
var aspnetMetrics = resourceMetrics.ScopeMetrics.Should().ContainSingle(x => x.Scope.Name == "OpenTelemetry.Instrumentation.AspNet").Which.Metrics;
aspnetMetrics.Should().ContainSingle(x => x.Name == "http.server.duration");
```


Because

```csharp
var aspnetScope = resourceMetrics.ScopeMetrics.Single(rm => rm.Scope.Name.Equals("OpenTelemetry.Instrumentation.AspNet", StringComparison.OrdinalIgnoreCase));
```

can throw an exception

Reply:
Was not aware of this feature. This makes it simpler.